server/protocols/protocols-smtp/src/test/java/org/apache/james/smtpserver/SMTPServerTest.java

```diff
@@ -492,6 +494,80 @@ public void testReceivedHeader() throws Exception {
             .isNotNull();
     }
 
+    @ParameterizedTest
+    @ValueSource(strings = {"good", "adomain.com", "sub.domain.com",  "127.0.0.1",  "[127.0.0.1]"})
```

Comment:
Do you mind adding an IPv6 in there ?

Reply:
sourced from https://en.wikipedia.org/wiki/IPv6_address
```
[fe80::1ff:fe23:4567:890a]
[2001:db8:85a3:8d3:1319:8a2e:370:7348]
```

Reply:
first one is a `link local address`
second one is reserved for documentation and example source code
if you want localhost it would be 
[::1]